pkgs/development/cuda-modules/setup-hooks/auto-add-opengl-runpath-hook.sh

```diff
@@ -9,7 +9,7 @@ elfHasDynamicSection() {
 autoAddOpenGLRunpathPhase() (
   local outputPaths
   mapfile -t outputPaths < <(for o in $(getAllOutputNames); do echo "${!o}"; done)
-  find "${outputPaths[@]}" -type f -executable -print0  | while IFS= read -rd "" f; do
+  find "${outputPaths[@]}" -type f -print0  | while IFS= read -rd "" f; do
```

Comment:
Here the `|` pipe can be replaced with `< <( find ...)` just as used in the line above.

Also, is `-d ""` actually correct? I saw people use `-d $'\0\` with `-print0` in other places. No idea how the syntax works but the intent is clear(er)

Reply:
I also wonder if we should consider adding at least some `-iname`, just to have fewer reads on average? Like `-iname '*.so*' -or '*.dylib*'`

Reply:
RE: `-iname`

Might as well delegate this to the user and add something like `${addDriverRunpath_findArgs-}`. A derivation we know is going to scan a ton of files of which very few are matches could just set some `addDriverRunpath_findArgs = [ "-executable" "-iname" "'*.so'" ]`

Reply:
Idk it's kind of neither here nor there: we're spawning subshells but we're nonetheless doing the bash loop instead of `-exec` because we want to invoke bash functions...  I'm fine with `find ... -exec`, and I put up with the `while ... done < <( ...)` loop for when the iteration body is pure bash (I hate the looks, but I pretend it's idiomatic and don't look). Am I a pessimist if I say we draw the worst from both worlds?

Reply:
I don't know, it's probably a matter of taste :slightly_smiling_face: the OCamler in me finds reverse application (aka `|`) left-to-right composition order more readable.

Though I agree that a `find -exec` would do as well here, and I see your point that piping to a while loop makes it a bit less readable; pipe is probably better for short commands without control-flow.